content/ADR-285-updated-friend-request.md

```diff
@@ -0,0 +1,631 @@
+
+rpc GetPendingFriendshipRequests(GetFriendshipRequestsPayload) returns
+(PaginatedFriendshipRequestsResponse) {}
+```
+
+#### Getting Sent Friend Requests
+
+```mermaid
+sequenceDiagram
+  participant U as User
+  participant SS as Social Service
+
+  U->>SS: GetSentFriendshipRequests(pagination: { limit: 50, offset: 0 })
+  SS-->>U: PaginatedFriendshipRequestsResponse with sent requests
+```
+
+```protobuf
+rpc GetSentFriendshipRequests(GetFriendshipRequestsPayload) returns
+(PaginatedFriendshipRequestsResponse) {}
+```
+
+### Getting Friends List
+
+Retrieving the user's current friends list is done through the `GetFriends` RPC call.
+
+```mermaid
+sequenceDiagram
+  participant U as User
+  participant SS as Social Service
+
+  U->>SS: GetFriends(pagination: { limit: 50, offset: 0 })
+  SS-->>U: PaginatedFriendsProfilesResponse with friends list
+```
+
+```protobuf
+message GetFriendsPayload {
+  optional Pagination pagination = 1;
+}
+
+message PaginatedFriendsProfilesResponse {
+  repeated FriendProfile friends = 1;
+  PaginatedResponse pagination_data = 2;
+}
+
+rpc GetFriends(GetFriendsPayload) returns (PaginatedFriendsProfilesResponse) {}
+```
+
+### Getting Mutual Friends
+
+Retrieving mutual friends between two users is done through
+the `GetMutualFriends` RPC call.
+
+```mermaid
+sequenceDiagram
+  participant U1 as User
+  participant SS as Social Service
+
+  U1->>SS: GetMutualFriends(user: "0x...", pagination: { limit: 50, offset: 0 })
+  SS-->>U1: PaginatedFriendsProfilesResponse with mutual friends
+```
+
+```protobuf
+message GetMutualFriendsPayload {
+  User user = 1;
+  optional Pagination pagination = 2;
+}
+
+rpc GetMutualFriends(GetMutualFriendsPayload)
+returns (PaginatedFriendsProfilesResponse) {}
+```
+
+### Getting Friendship Status
+
+Checking the friendship status between two users is done by performing the
+`GetFriendshipStatus` RPC call.
+
+```mermaid
+sequenceDiagram
+  participant U1 as User
+  participant SS as Social Service
+
+  U1->>SS: GetFriendshipStatus(user: "0x...")
+  SS-->>U1: GetFriendshipStatusResponse with status
+```
+
+```protobuf
+message GetFriendshipStatusPayload {
+  User user = 1;
+}
+
+enum FriendshipStatus {
+  REQUEST_SENT = 0;
+  REQUEST_RECEIVED = 1;
+  CANCELED = 2;
+  ACCEPTED = 3;
+  REJECTED = 4;
+  DELETED = 5;
+  BLOCKED = 6;
+  NONE = 7;
+  BLOCKED_BY = 8;
+}
+
+message GetFriendshipStatusResponse {
+  message Ok {
+    FriendshipStatus status = 1;
+    optional string message = 2;
+  }
+
+  oneof response {
+    Ok accepted = 1;
+    InternalServerError internal_server_error = 2;
+    InvalidRequest invalid_request = 3;
+  }
+}
+
+rpc GetFriendshipStatus(GetFriendshipStatusPayload) returns
+(GetFriendshipStatusResponse) {}
+```
+
+### Subscribing to Friendship Updates
+
+Subscribing to friendship updates is critical for real-time friend request
+management. This subscription is done by performing the
+`SubscribeToFriendshipUpdates` RPC call to the Social Service.
+
+```mermaid
```

Comment:
This mermaid does not compile and fails in the rendering, the multiline message block needs to be in a single line 
`SS-->>U2: FriendshipUpdate(request: { friend: {...}, created_at: 123, message: "Hello!", id: "...", status: REQUEST_SENT }),`